packages/tasit-action/src/Contract.js

```diff
@@ -84,18 +107,23 @@ export class Contract {
 
   #attachReadFunction = f => {
     this[f.name] = async (...args) => {
-      const value = await this.#contract[f.name].apply(null, args);
+      const value = await this.#ethersContract[f.name].apply(null, args);
       return value;
     };
   };
 
   #attachWriteFunction = f => {
     this[f.name] = (...args) => {
-      if (!Utils.isEthersJsSigner(this.#contract.signer))
+      if (!Utils.isEthersJsSigner(this.#ethersContract.signer))
         throw new Error(`Cannot write data to a Contract without a wallet`);
 
-      const tx = this.#contract[f.name].apply(null, args);
-      const subscription = new TransactionSubscription(tx, this.#provider);
+      const tx = this.#ethersContract[f.name].apply(null, args);
+
+      const subscription = new TransactionSubscription(
+        tx,
+        this.#provider,
+        this
```

Comment:
Yes, I've seen that. I've kept the `this.#provider` arg because I'm not sure if we want the function `Contract.getProvider()` (for now it is as "testing purpose function"). Maybe rename to `_getProvider()` ?

Reply:
A) I guess having the option of using different providers for different actions could be nice, so in that way it's nice having it as an explicit param.

B) Maybe we add the ability to do `new TransactionSubscriptionWithPresetProvider(...)` in the future, and that uses this under the hood.

C) Wait, I don't totally follow why the `this.#provider` arg is affected by whether or not we have `Contract.getProvider()`

D) And sure, renaming to the underscore could make sense.

Reply:
Agreed that this may be deprecated based on the other discussion.

C) I get it now. So we either (1) pass just the contract and tx as args and use the get function on the contract to get the provider (and thus use it not just for testing the contract itself). Or (2) we pass just the provider and tx as args, and see if we can make do without passing the contract.

Reply:
D) Done
C) Still open

Reply:
C) Done

Reply:
Great! Resolving this.